src/messaging/ReliableMessageProtocolConfig.h

```diff
@@ -129,7 +129,7 @@ namespace chip {
 #ifndef CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE
 #if CHIP_SYSTEM_CONFIG_USE_LWIP
 
-#if !LWIP_PBUF_FROM_CUSTOM_POOLS && PBUF_POOL_SIZE != 0
+#if !(defined(LWIP_PBUF_FROM_CUSTOM_POOLS) && LWIP_PBUF_FROM_CUSTOM_POOLS) && PBUF_POOL_SIZE != 0
```

Comment:
This is not right.  `#include <lwip/opt.h>` above pulls in a definition of LWIP_PBUF_FROM_CUSTOM_POOLS, no?

Reply:
`#include <lwip/opt.h>` includes `lwipopts.h` and ESP32 uses the lwip from the esp-idf which does not contains the config `LWIP_PBUF_FROM_CUSTOM_POOLS`.

`third_party/lwip/repo/lwip` was added two years back and the one in esp-idf is pretty new so, this option could have been removed in newer lwip.

Reply:
Ok, well, we need to figure out how to handle this case so we detect failure to include lwip/opt.h...

Is there a new thing that replaced LWIP_PBUF_FROM_CUSTOM_POOLS?  Is there something smart we can do about checking for another define that would always come from there assuming that it's included?

Reply:
I have added a new config option `CHIP_SYSTEM_CONFIG_LWIP_PBUF_FROM_CUSTOM_POOL` which checks if `LWIP_PBUF_FROM_CUSTOM_POOLS` is present and if not, it defines it with value `0`.